### PR TITLE
[CI] Use gcc-10 for inductor tests

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100262

Summary: This is needed to run torchrec_dlrm. See https://github.com/pytorch/benchmark/pull/1427

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @soumith

### CODE REVIEW

File: .github/ci_commit_pins/audio.txt [hunk 1]
```diff
@@ -1 +1 @@
-a8f4e97bd5356a7a77510cdf6a3a62e25a5dc602
\ No newline at end of file
+151ac4d85007c7e14d9ece9023a2c2b4b0cc6a40
```
Comment (malfet) [inline .github/ci_commit_pins/audio.txt:1]: Why it needs an audio pin?

**Review by malfet — CHANGES_REQUESTED** [review]
Rather than adding a new config, do you mind just migrating `pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc7` to `pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc11`?

File: .github/ci_commit_pins/audio.txt (hunk 1 above)
Comment (desertfire) [reply .github/ci_commit_pins/audio.txt:1]: I am trying to solve an issue for running torchrec_dlrm in torchbench, and this is required, at least when I tried it locally.

**Comment by desertfire** [comment]
> Rather than adding a new config, do you mind just migrating `pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc7` to `pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc11`?

I am not sure this PR will fix the problem I am solving. Will follow your recommendation if it work.

**Review by huydhn — APPROVED** [review]
LGTM!  As there is a new Docker image, I vaguely remember that we need to create an entry for it on ECR.  Let's see if the build works

**Comment by desertfire** [comment]
> LGTM! As there is a new Docker image, I vaguely remember that we need to create an entry for it on ECR. Let's see if the build works

OK, this doesn't work. @huydhn , what is ECR?

**Comment by huydhn** [comment]
> OK, this doesn't work. @huydhn , what is ECR?

Ah, that's where the Docker images are pushed to, in this case, it's `308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc11`, so you can see the path `/pytorch/pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc11` is expected to be there.  Let me create it for you and retry the job

**Review by malfet — APPROVED** [review]

**Review by ZainRizvi — APPROVED** [review]

**Comment by malfet** [comment]
> @malfet @huydhn @ZainRizvi , I found the particular missing symbol issue for torchrec_dlrm relates to GLIBCXX_USE_CXX11_ABI. When we build from source, GLIBCXX_USE_CXX11_ABI is set to 1 according to https://github.com/pytorch/pytorch/blob/main/torch/abi-check.cpp. When we install from nightly, GLIBCXX_USE_CXX11_ABI is set to 0, and fbgemm-gpu-nightly is built based on pytorch nightly which means it is built with GLIBCXX_USE_CXX11_ABI=0, thus causing symbol mismatch.
> 
> Is GLIBCXX_USE_CXX11_ABI=0 for pytorch nightly the expected behavior? How can we resolve this kind of mismatch behavior?

You can build `torchrec_dlrm` from source, or you can change CI builds to use `GLIBCXX_USE_CXX1_ABI=0`

**Comment by desertfire** [comment]
> @desertfire yes this is the expected behavior.

I guess it's because of portability concern?

> And to solve it guess you can either build `torchrec_dlrm` from source(as we do with the rest of the deps), or you can change CI builds to use `GLIBCXX_USE_CXX1_ABI=0`

FBGEMM does not provide a convenient way to pip install from source, so I will try building CI with `GLIBCXX_USE_CXX11_ABI =0` first.

**Comment by desertfire** [comment]
@malfet BTW, I found that torchaudio has an inconsistent behavior in its nightly build, from the following error msg when I tried with locally-installed nightly torch/torchaudio/torchvision/torchtext. We should make the `GLIBCXX_USE_CXX11_ABI` choice consistent across our libs. 

```
torchaudio.so: undefined symbol: _ZNK5torch8autograd4Node4nameB5cxx11Ev
```

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**